itwm_example/mco/tests/test_mco.py

```diff
@@ -38,124 +30,56 @@ def setUp(self):
-            [0.50, 0.0, 0.50],
-            [0.25, 0.75, 0.0],
-            [0.25, 0.50, 0.25],
-            [0.25, 0.25, 0.50],
-            [0.25, 0.0, 0.75],
-            [0.0, 1.0, 0.0],
-            [0.0, 0.75, 0.25],
-            [0.0, 0.50, 0.50],
-            [0.0, 0.25, 0.75],
-            [0.0, 0.0, 1.0],
-        ])
-
-        self.assertEqual(
-            list(get_weight_combinations(3, 9)),
-            [[1.0, 0.0, 0.0], [0.875, 0.125, 0.0], [0.875, 0.0, 0.125],
-             [0.75, 0.25, 0.0], [0.75, 0.125, 0.125], [0.75, 0.0, 0.25],
-             [0.625, 0.375, 0.0], [0.625, 0.25, 0.125], [0.625, 0.125, 0.25],
-             [0.625, 0.0, 0.375], [0.5, 0.5, 0.0], [0.5, 0.375, 0.125],
-             [0.5, 0.25, 0.25], [0.5, 0.125, 0.375], [0.5, 0.0, 0.5],
-             [0.375, 0.625, 0.0], [0.375, 0.5, 0.125], [0.375, 0.375, 0.25],
-             [0.375, 0.25, 0.375], [0.375, 0.125, 0.5], [0.375, 0.0, 0.625],
-             [0.25, 0.75, 0.0], [0.25, 0.625, 0.125], [0.25, 0.5, 0.25],
-             [0.25, 0.375, 0.375], [0.25, 0.25, 0.5], [0.25, 0.125, 0.625],
-             [0.25, 0.0, 0.75], [0.125, 0.875, 0.0], [0.125, 0.75, 0.125],
-             [0.125, 0.625, 0.25], [0.125, 0.5, 0.375], [0.125, 0.375, 0.5],
-             [0.125, 0.25, 0.625], [0.125, 0.125, 0.75], [0.125, 0.0, 0.875],
-             [0.0, 1.0, 0.0], [0.0, 0.875, 0.125], [0.0, 0.75, 0.25],
-             [0.0, 0.625, 0.375], [0.0, 0.5, 0.5], [0.0, 0.375, 0.625],
-             [0.0, 0.25, 0.75], [0.0, 0.125, 0.875], [0.0, 0.0, 1.0]]
-        )
+        optimizer = MockOptimizer(self.evaluator, self.parameters)
 
-        self.assertEqual(list(get_weight_combinations(2, 5, False)), [
-            [0.75, 0.25],
-            [0.50, 0.50],
-            [0.25, 0.75],
-        ])
-
-        self.assertEqual(
-            list(get_weight_combinations(3, 9, False)),
-            [[0.75, 0.125, 0.125],
-             [0.625, 0.25, 0.125],
-             [0.625, 0.125, 0.25],
-             [0.5, 0.375, 0.125],
-             [0.5, 0.25, 0.25],
-             [0.5, 0.125, 0.375],
-             [0.375, 0.5, 0.125],
-             [0.375, 0.375, 0.25],
-             [0.375, 0.25, 0.375],
-             [0.375, 0.125, 0.5],
-             [0.25, 0.625, 0.125],
-             [0.25, 0.5, 0.25],
-             [0.25, 0.375, 0.375],
-             [0.25, 0.25, 0.5],
-             [0.25, 0.125, 0.625],
-             [0.125, 0.75, 0.125],
-             [0.125, 0.625, 0.25],
-             [0.125, 0.5, 0.375],
-             [0.125, 0.375, 0.5],
-             [0.125, 0.25, 0.625],
-             [0.125, 0.125, 0.75]]
-        )
+        scaling_factors = self.mco.get_scaling_factors(optimizer, temp_kpis)
+
+        self.assertEqual(scaling_factors, [0.1, 1.0])
+
+    def test__space_search_distribution(self):
+        for strategy, klass in (
+            ("Uniform", UniformSpaceSampler),
+            ("Dirichlet", DirichletSpaceSampler),
+            ("Uniform", UniformSpaceSampler),
```

Comment:
Are we testing anything extra by including the `UniformSpaceSampler` twice here?

Reply:
Yes, my idea was to test the setup of the `sampler` type, which means I want to make sure it is properly set up after multiple changes.